core/tee/tee_rpmb_fs.c

```diff
@@ -2410,20 +2414,26 @@ static TEE_Result update_write_helper(struct rpmb_file_handle *fh,
 		/* Possibly update data in temporary buffer */
 		if ((blk_offset + TMP_BLOCK_SIZE > pos) &&
 		    (blk_offset < pos + size)) {
-			uint8_t *dst = blk_buf;
-			size_t copy_size = TMP_BLOCK_SIZE;
+			size_t offset = 0;
+
+			copy_dst = blk_buf;
+			copy_size = TMP_BLOCK_SIZE;
 
 			if (blk_offset < pos) {
-				size_t offset = pos - blk_offset;
+				offset = pos - blk_offset;
 
-				dst += offset;
+				copy_dst += offset;
 				copy_size -= offset;
 			}
 			copy_size = MIN(copy_size, rem_size);
 
-			memcpy(dst, rem_buf, copy_size);
+			memcpy(copy_dst, rem_buf, copy_size);
 			rem_buf += copy_size;
 			rem_size -= copy_size;
+
+			memset(blk_buf + rd_size, 0, offset - rd_size);
```

Comment:
Please add a comment explaining the `memset()` strategy here. It is quite hard to figure out what's going on here.

Reply:
ok